### ISSUE TITLE
SQLSTATE[23000]

### ISSUE THREAD

**Issue by untuned** [NamelessMC/Nameless#1082]
A user failed to register due to this error:
`Verification failed, please try again. - SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'password' cannot be null`
I have MCAssoc enabled.

**Comment by samerton** [comment]
Are you using Nameless version 1 or the version 2 pre-releases (you can double check in the AdminCP -> Overview tab)?

**Comment by untuned** [comment]
Nameless 2.0.0 pre-4

**Comment by samerton** [comment]
Thanks, fixed in 02da10a5cc09a6cc94cbb6abc548b025af0b9c00, which will be included in the next update.

If you want to apply the fix now, you can apply the change from the commit by replacing
```php
if((isset($_SESSION['remember']) || isset($_SESSION['username']) || isset($_SESSION['email']) || isset($_SESSION['password'])) && !isset($_POST['tfa_code'])){
```

with

```php
if((isset($_SESSION['remember']) || isset($_SESSION['username']) || isset($_SESSION['email']) || isset($_SESSION['password'])) && (!isset($_POST['tfa_code']) && !isset($_SESSION['mcassoc']))){
```

in **core/init.php**
